packages/rpc-core/src/rpc-methods/getTransaction.ts

```diff
@@ -0,0 +1,254 @@
+
+type ReturnData = {
+    /** the program that generated the return data */
+    programId: Base58EncodedAddress;
+    /** the return data itself */
+    data: Base64EncodedDataResponse;
+};
+
+type TransactionMetaBase = Readonly<{
+    /** Error if transaction failed, null if transaction succeeded. */
+    err: TransactionError | null;
+    /** fee this transaction was charged */
+    fee: LamportsUnsafeBeyond2Pow53Minus1;
+    /** array of account balances from before the transaction was processed */
+    preBalances: readonly U64UnsafeBeyond2Pow53Minus1[];
+    /** array of account balances after the transaction was processed */
+    postBalances: readonly U64UnsafeBeyond2Pow53Minus1[];
+    /** List of token balances from before the transaction was processed or omitted if token balance recording was not yet enabled during this transaction */
+    preTokenBalances?: readonly TokenBalance[];
+    /** List of token balances from after the transaction was processed or omitted if token balance recording was not yet enabled during this transaction */
+    postTokenBalances?: readonly TokenBalance[];
+    /** array of string log messages or null if log message recording was not enabled during this transaction */
+    logMessages: readonly string[] | null;
+    /** transaction-level rewards */
+    rewards: readonly TransactionReward[] | null;
+    /**
+     * Transaction status
+     * @deprecated
+     */
+    status: TransactionStatus;
+    /** the most-recent return data generated by an instruction in the transaction */
+    returnData?: ReturnData;
+    /** number of compute units consumed by the transaction */
+    computeUnitsConsumed?: U64UnsafeBeyond2Pow53Minus1;
+}>;
+
+type AddressTableLookup = Readonly<{
+    /** public key for an address lookup table account. */
+    accountKey: Base58EncodedAddress;
+    /** List of indices used to load addresses of writable accounts from a lookup table. */
+    writableIndexes: readonly number[];
+    /** List of indices used to load addresses of readonly accounts from a lookup table. */
+    readableIndexes: readonly number[];
+}>;
+
+type TransactionBase = Readonly<{
+    message: {
+        recentBlockhash: Blockhash;
+    };
+    signatures: readonly Base58EncodedBytes[];
+}>;
+
+type TransactionInstruction = Readonly<{
+    programIdIndex: number;
+    accounts: readonly number[];
+    data: Base58EncodedBytes;
+}>;
+
+type TransactionJson = TransactionBase &
+    Readonly<{
+        message: {
+            accountKeys: readonly Base58EncodedAddress[];
+            header: {
+                numReadonlySignedAccounts: number;
+                numReadonlyUnsignedAccounts: number;
+                numRequiredSignatures: number;
+            };
+            instructions: readonly TransactionInstruction[];
+        };
+    }>;
+
+type TransactionInstructionNoParser = Readonly<{
+    accounts: readonly Base58EncodedAddress[];
+    data: Base58EncodedBytes;
+    programId: Base58EncodedAddress;
+}>;
```

Comment:
Absolutely nuts.

The [docs](https://docs.solana.com/api/http#getaccountinfo) are also wrong about this, right? https://github.com/solana-labs/solana/issues/31701

> If jsonParsed is requested but a parser cannot be found, the instruction falls back to regular JSON encoding (accounts, data, and programIdIndex fields).

I [traced through the code](https://github.com/solana-labs/solana/blob/bd8289e1149fa412220215c5825089eecc413c0b/transaction-status/src/lib.rs#LL151C9-L155C10). I believe this is the `UiPartiallyDecodedInstruction` type. Rename maybe?

```suggestion
type PartiallyDecodedTransactionInstruction = Readonly<{
    accounts: readonly Base58EncodedAddress[];
    data: Base58EncodedBytes;
    programId: Base58EncodedAddress;
}>;
```

Reply:
Yep agreed, this doesn't match the docs. Really confusing to reason about too!